Tests/Unit/NetworkServer/CupsUpdateInfoResponseHeaderTests.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.Tests.Unit.NetworkServer
+{
+    using System;
+    using System.Buffers;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using LoRaWan.NetworkServer.BasicsStation;
+    using Xunit;
+
+    public class CupsUpdateInfoResponseHeaderTests
+    {
+        private readonly string updateUriString = "https://localhost:1234";
+
+        [Fact]
+        public void Serialize_WithNoUpdates()
+        {
+            var updateResponseHeader = new CupsUpdateInfoResponseHeader
+            {
+                SignatureKeyCrc = 0,
+                UpdateDataLength = 0,
+            };
+
+            using var memoryPool = MemoryPool<byte>.Shared.Rent(2048);
```

Comment:
Consider renaming `memoryPool` to `memoryRental` since it's not the pool but the rented memory from the pool.

Reply:
I would even skip the memory pool and just create an array (`new byte[2048]`). This is a test and we don't care about memory pooling.

Reply:
Renamed, but did not get rid of the array creation